diff --git a/Makefile b/Makefile
index 1e88984..f2499f5 100644
--- a/Makefile
+++ b/Makefile
@@ -128,6 +128,7 @@ mkfs/mkfs: mkfs/mkfs.c $K/fs.h $K/param.h
 UPROGS=\
 	$U/_cat\
 	$U/_echo\
+	$U/_hello\
 	$U/_forktest\
 	$U/_grep\
 	$U/_init\
diff --git a/kernel/syscall.c b/kernel/syscall.c
index 822720c..d790a71 100644
--- a/kernel/syscall.c
+++ b/kernel/syscall.c
@@ -102,6 +102,7 @@ extern uint64 sys_unlink(void);
 extern uint64 sys_link(void);
 extern uint64 sys_mkdir(void);
 extern uint64 sys_close(void);
+extern uint64 sys_hello(void);
 
 // An array mapping syscall numbers from syscall.h
 // to the function that handles the system call.
@@ -128,6 +129,7 @@ static uint64 (*syscalls[])(void) = {
   [SYS_link]    sys_link,
   [SYS_mkdir]   sys_mkdir,
   [SYS_close]   sys_close,
+  [SYS_hello]   sys_hello,
   // clang-format on
 };
 
diff --git a/kernel/syscall.h b/kernel/syscall.h
index 454d1b4..57abe5b 100644
--- a/kernel/syscall.h
+++ b/kernel/syscall.h
@@ -20,3 +20,4 @@
 #define SYS_link   19
 #define SYS_mkdir  20
 #define SYS_close  21
+#define SYS_hello  22
diff --git a/kernel/sysproc.c b/kernel/sysproc.c
index 73ff1f3..f76ffe1 100644
--- a/kernel/sysproc.c
+++ b/kernel/sysproc.c
@@ -107,3 +107,9 @@ sys_uptime(void)
   release(&tickslock);
   return xticks;
 }
+
+uint64
+sys_hello(void)
+{
+  return 2026;
+}
diff --git a/user/hello.c b/user/hello.c
new file mode 100644
index 0000000..7b624a7
--- /dev/null
+++ b/user/hello.c
@@ -0,0 +1,10 @@
+#include "kernel/types.h"
+#include "user/user.h"
+
+int
+main(void)
+{
+  int value = hello();
+  printf("hello syscall returned %d\n", value);
+  exit(0);
+}
diff --git a/user/user.h b/user/user.h
index ecd0921..de76c8c 100644
--- a/user/user.h
+++ b/user/user.h
@@ -24,6 +24,7 @@ int getpid(void);
 char *sys_sbrk(int, int);
 int pause(int);
 int uptime(void);
+int hello(void);
 
 // ulib.c
 int stat(const char *, struct stat *);
diff --git a/user/usys.pl b/user/usys.pl
index c5d4c3a..ef380bf 100755
--- a/user/usys.pl
+++ b/user/usys.pl
@@ -42,3 +42,4 @@ entry("getpid");
 entry("sbrk");
 entry("pause");
 entry("uptime");
+entry("hello");
